arch/lkl/kernel/setup.c

```diff
@@ -112,7 +112,8 @@ void machine_restart(char *unused)
 long lkl_sys_halt(void)
 {
 	long err;
-	long params[6] = { 0, };
+	long params[6] = {LINUX_REBOOT_MAGIC1,
+		LINUX_REBOOT_MAGIC2, LINUX_REBOOT_CMD_RESTART, };
```

Comment:
Looking at the kernel/reboot.c, LINUX_REBOOT_CMD_HALT will call do_exit which is not what we want.